src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -382,7 +382,7 @@ public async Task WriteSummaryAsync(NuGetOperationStatus operationStatus, TimeSp
             Func<RestoreOperationLogger, RestoreOperationProgressUI, CancellationToken, Task> asyncRunMethod,
             CancellationToken token)
         {
-            using (var progress = await _progressFactory(token))
+            await using (var progress = await _progressFactory(token))
```

Comment:
@nkolev92 I think you'll want to think about if you want to block the whole operation here based on sending data to the UI thread. If its busy that might be a while.